File: GeeksForGeeks/DSA_Series/Day1-Arrays/p1.py
## Equilibrium Point

def equilibriumPoint(A, N):
    # Your code here
    total = sum(A)
    
    i=0
    rtot = 0
    while i<N:
        if rtot == total-A[i]-rtot:
            return i+1
        rtot+=A[i]
        i+=1
    return -1

test_cases = [([1,3,5,2,2],5), ([1],1),([1,3,1],3),([1,3,4,9,6,1,1],7),([1,3,4,9,6,1],6)]
for x,y in test_cases:
    print(equilibriumPoint(x,y))

File: GeeksForGeeks/DSA_Series/Day5-SlidingWindow/p1.cpp
// Subarray with given sum
#include<iostream>
#include<bits/stc++.h>

vector<int> subarraySum(int arr[], int n, long long s)
{
    // Your code here
    int currSum = arr[0], idx=1, begin=0;
    vector<int> result;
    while(idx<=n){
        while(currSum>s && begin<idx-1){
            currSum = currSum-arr[begin];
            begin++;
        }
        
        if(currSum==s){
            result.push_back(begin+1);
            result.push_back(idx);
            return result;
        }
        if(idx<n){
            currSum = currSum + arr[idx];
        }
        idx++;
    }
    result.push_back(-1);
    return result;
};